### PR TITLE
Indicate Dockerfile can be used, simplify now that image metadata is available

### PR DESCRIPTION
**Pull request by Chuxel**
This PR addresses several things in existing templates to dramatically simplify and clarify them.

1. It adds in a reference to a to-be-written guide that explains how image vs Dockerfile vs Docker Compose usage when just an `image` is referenced.

1. It removes Features and settings where they are now in the actual referenced image's metadata. This capability is new, so not all templates have been updated.
    1. It also removes the common script being in the docker and kubernetes templates since any reference to a custom image has been removed.
    1. It also removes the `java-8` template since it is no longer needed given the difference was pure VS Code settings, which are now on the image.
    
3. It removes a couple of "mount" references that points to docker-from-docker given the Feature can now handle this automatically.

4. It takes advantage of the _REMOTE_USER environment variable in the Kubernetes templates now that this is available (and was not when it was created).
 
4. It drops .NET 3.1 from lists given it is now out of support.
 
6. It fixes comments in the "existing Dockerfile" and "existing Docker Compose" scenarios that incorrectly mentioned un-commenting to access root. Given these are customer images, there's no garuntee image metadata for remoteUser is present.

### CODE REVIEW

File: src/rust-postgres/.devcontainer/devcontainer.json [hunk 1]
```diff
@@ -4,42 +4,20 @@
 	"name": "Rust and PostgreSQL",
 	"dockerComposeFile": "docker-compose.yml",
 	"service": "app",
-	"workspaceFolder": "/workspaces/${localWorkspaceFolderBasename}",
+	"workspaceFolder": "/workspaces/${localWorkspaceFolderBasename}"
```
Comment (samruddhikhandale) [inline src/rust-postgres/.devcontainer/devcontainer.json:7]: Same comment here. Version bump reminder 👀 ?

File: src/rust/.devcontainer/devcontainer.json [hunk 2]
```diff
@@ -2,6 +2,7 @@
 // README at: https://github.com/devcontainers/templates/tree/main/src/rust
 {
 	"name": "Rust",
```
Comment (samruddhikhandale) [inline src/rust/.devcontainer/devcontainer.json:4]: Version bump reminder 👀 ?

**Comment by Chuxel** [comment]
@samruddhikhandale Updated!

**Review by samruddhikhandale — APPROVED** [review]
Looks great, thanks for putting this up.
Feel free to merge when https://containers.dev/guide/dockerfile is live.

**Pull request merged**